app/js/components/common/Header.jsx

```diff
@@ -48,7 +57,15 @@ export class Header extends Component {
         this.setState({currentLocationTag: e.target.id});
     }
 
-    dropDownMenu(locationTags) {
+    setLocation(e, locationUUIDTag) {
+        this.setState({currentLocationTag: e.target.id});
+        const apiHelper = new ApiHelper(null);
+        apiHelper.post("/appui/session", {"location" : locationUUIDTag}).then(response => {
+            // nothing to do
```

Comment:
It's better to leave a comment like 'Re-render the location based components in UI'

Reply:
Changed